cmake/CMakeLists.txt

```diff
@@ -1057,6 +1057,9 @@ function(onnxruntime_set_compile_flags target_name)
     foreach(FLAG ${ORT_WARNING_FLAGS})
       target_compile_options(${target_name} PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${FLAG}>")
     endforeach()
+    if("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_LESS 13 AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 12)
+      target_compile_options(${target_name} PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:-Wno-maybe-uninitialized>")
```

Comment:
can this disabled warning be localized? e.g., if it's only from absl/container/flat_hash_set.h, and we only include that header through include/onnxruntime/core/common/inlined_containers.h, can we just disable it around that include in inlined_containers.h?

Reply:
This is temporary.  Soon I will upgrade GCC to 13, then we don't need to worry about it.

Reply:
Though the log says the error is from inlined_containers.h, but actually it's hard to find out which line of code caused the issue. I tried to compile the following code alone and it didn't cause any trouble. 
```
AxesSet unique_axes
```
Then I realized it was probably just an internal compiler bug.